Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/GlobalLocalAnomalyDetector.java

```diff
@@ -211,19 +214,35 @@ public GenericAnomalyDescriptor<P> process(P object, float weight, BiFunction<P,
                         globalDistance, null);
             }
         }
-        Weighted<P> result = score(object, localDistance, considerOcclusion);
-
+        List<Weighted<P>> result = score(object, localDistance, considerOcclusion);
         double threshold = thresholder.threshold();
         double grade = 0;
-        if (result.weight > 0) {
-            grade = thresholder.getAnomalyGrade(result.weight, false);
-        }
+        float score = 0;
+        if (result != null) {
+            score = result.stream().map(a -> a.weight).reduce(FLOAT_MAX, Float::min);
+            if (score < FLOAT_MAX) {
+                // an exponential attribution
+                double sum = result.stream()
+                        .map(a -> (double) ((a.weight == FLOAT_MAX) ? 0 : exp(-a.weight * a.weight)))
```

Comment:
Could you explain the reason of choosing an exponential function over other functions like a harmonic mean?